src/main/java/edu/ucsb/cs156/gauchoride/errors/IllegalRequestException.java

```diff
@@ -0,0 +1,7 @@
+package edu.ucsb.cs156.gauchoride.errors;
+
+public class IllegalRequestException extends RuntimeException {

```

Comment:
Right now when the exception is triggered the status code is 500, but realistically I think it should be 400 because the request is invalid and malformed.

Reply:
```suggestion
package edu.ucsb.cs156.gauchoride.errors;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

@ResponseStatus(HttpStatus.BAD_REQUEST)
public class IllegalRequestException extends RuntimeException {

```

Reply:
I agree with you because it would be trigger when a user creates a ride request with no rider which is a must in the ride request because then who is going to drive the rider? Hence, it should be 400 because the create request cannot be process
since it is some user error.